presto-main/src/main/java/com/facebook/presto/operator/scalar/RowToJsonCast.java

```diff
@@ -72,19 +74,21 @@ public BuiltInScalarFunctionImplementation specialize(BoundVariables boundVariab
 
         List<Type> fieldTypes = type.getTypeParameters();
         List<JsonGeneratorWriter> fieldWriters = new ArrayList<>(fieldTypes.size());
+        List<TypeSignatureParameter> typeSignatureParameters = type.getTypeSignature().getParameters();
+        List<String> fieldNames = new ArrayList<>(fieldTypes.size());
         for (int i = 0; i < fieldTypes.size(); i++) {
```

Comment:
Can we do something like below? 

List<String> fieldNames = IntStream.range(0, fieldTypes.size())
    .mapToObj(i -> typeSignatureParameters.get(i).getNamedTypeSignature().getName().orElse(""))
    .collect(Collectors.toList());

IntStream.range(0, fieldTypes.size())
    .forEach(i -> fieldWriters.add(createJsonGeneratorWriter(fieldTypes.get(i))));